python311Packages.linear-garage-door: init at 0.2.9

fabaff opened the pull request:
## Description of changes

Control Linear Garage Doors with Python

https://github.com/IceBotYT/linear-garage-door


<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fabaff commented:
Waiting for #288233 to be merged to keep the `component-packages.nix` update simple.

fabaff commented:
Result of `nixpkgs-review pr 288329` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.linear-garage-door</li>
    <li>python311Packages.linear-garage-door.dist</li>
    <li>python312Packages.linear-garage-door</li>
    <li>python312Packages.linear-garage-door.dist</li>
  </ul>
</details>

SuperSandro2000 commented on pkgs/development/python-modules/linear-garage-door/default.nix:
```diff
@@ -0,0 +1,54 @@
+  };
+
+  postPatch = ''
+    sed -i "/--cov/d" pyproject.toml
+    sed -i "/--no-cov/d" pyproject.toml
```
```suggestion
    sed -i pyproject.toml \
      -e "/--cov/d" \
      -e "/--no-cov/d"
```
we can combine this

fabaff replied:
Thanks, updated.

SuperSandro2000 commented on pkgs/development/python-modules/linear-garage-door/default.nix:
```diff
@@ -0,0 +1,54 @@
+    tenacity
+  ];
+
+  # Module doesn't have tests
```
yeah, you can't call this tests https://github.com/IceBotYT/linear-garage-door/tree/main/tests

fabaff replied:
As the boilerplate is there, I guess that with one of the next releases there will be tests.

fabaff committed:
python311Packages.linear-garage-door: init at 0.2.9

Control Linear Garage Doors with Python

https://github.com/IceBotYT/linear-garage-door

fabaff committed:
home-assistant: update component-packages

fabaff commented:
Result of `nixpkgs-review pr 288329` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>home-assistant-component-tests.linear_garage_door</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.linear-garage-door</li>
    <li>python311Packages.linear-garage-door.dist</li>
    <li>python312Packages.linear-garage-door</li>
    <li>python312Packages.linear-garage-door.dist</li>
  </ul>
</details>

fabaff commented:
Result of `nixpkgs-review pr 288329` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.linear_garage_door</li>
    <li>python311Packages.linear-garage-door</li>
    <li>python311Packages.linear-garage-door.dist</li>
    <li>python312Packages.linear-garage-door</li>
    <li>python312Packages.linear-garage-door.dist</li>
  </ul>
</details>
